perllib/FixMyStreet/App/Controller/Report.pm

```diff
@@ -521,6 +521,20 @@ sub inspect : Private {
             }
         }
 
+        # set assignment
+        my $assigned = ($c->get_param('assignment'));
+        if ($assigned && $assigned eq 'unassigned') {
+            # take off shortlist...
```

Comment:
This should use the standard `user->remove_from_planned_reports(report)` function, unless there is a reason it is not that's not mentioned.